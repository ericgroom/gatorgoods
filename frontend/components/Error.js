import styled from "styled-components";

const ErrorWrapper = styled.div`
    background-color: #FA8072;
    color: #960018;
    padding: 0.5rem;
    border-radius: 2rem;
    text-align: center;
    display: flex;
    justify-content: center;
    align-content: center;
`;

const Error = props => (
    <>
    { props.error && <ErrorWrapper>
        {props.error.message}
    </ErrorWrapper>
    }
    </>
);

export default Error;