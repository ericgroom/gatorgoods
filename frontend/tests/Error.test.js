/*
*   Tests for proper function of error.js
*/

import Error from "../components/Error";
import { shallow } from "enzyme";

describe("Error component", () => {
  it("doesn't render when passed null", () => {
    const error = shallow(<Error />);
    expect(error).toMatchSnapshot();
    expect(error.children()).toHaveLength(0);
  });
  it("renders when passed an error message", () => {
    const error = shallow(<Error error={{ message: "An Error" }} />);
    expect(error).toMatchSnapshot();
    expect(error.children()).not.toHaveLength(0);
  });
});
