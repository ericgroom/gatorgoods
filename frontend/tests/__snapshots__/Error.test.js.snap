// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component doesn't render when passed null 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Error component renders when passed an error message 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    error={
      Object {
        "message": "An Error",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "An Error",
    },
    "ref": null,
    "rendered": "An Error",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Error__ErrorWrapper-a71vg1-0",
        "isStatic": true,
        "rules": Array [
          "background-color:#fa8072;color:#960018;padding:0.5rem;border-radius:2rem;text-align:center;display:flex;justify-content:center;align-content:center;",
        ],
      },
      "displayName": "Error__ErrorWrapper",
      "render": [Function],
      "styledComponentId": "Error__ErrorWrapper-a71vg1-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "An Error",
      },
      "ref": null,
      "rendered": "An Error",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "Error__ErrorWrapper-a71vg1-0",
          "isStatic": true,
          "rules": Array [
            "background-color:#fa8072;color:#960018;padding:0.5rem;border-radius:2rem;text-align:center;display:flex;justify-content:center;align-content:center;",
          ],
        },
        "displayName": "Error__ErrorWrapper",
        "render": [Function],
        "styledComponentId": "Error__ErrorWrapper-a71vg1-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
